File: spring/src/main/java/com/david/hlp/Spring/repeater/module/entity/Project.java
package com.david.hlp.Spring.repeater.module.entity;

import java.util.Date;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * 项目信息实体类
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Project {
    /**
     * 主键ID
     */
    private Long id;

    /**
     * 项目名称
     */
    private String projectName;

    /**
     * 文档说明
     */
    private String doc;

    /**
     * 创建时间
     */
    private Date gmtCreate;

    /**
     * 修改时间
     */
    private Date gmtModified;

    /**
     * 是否删除:0-未删除,1-已删除
     */
    private Integer isDeleted;
}

File: vue/src/types/repeater/project.ts
import type { RoleUrl } from './roleurl'

/**
 * 项目信息接口
 */
export interface Project {
  /**
   * 主键ID
   */
  id: number | string

  /**
   * 项目名称
   */
  projectName?: string

  /**
   * 文档说明
   */
  doc?: string

  /**
   * 创建时间
   */
  gmtCreate?: string

  /**
   * 修改时间
   */
  gmtModified?: string

  /**
   * 是否删除:0-未删除,1-已删除
   */
  isDeleted?: number
}

/**
 * 项目角色DTO
 */
export interface ProjectRoleDTO extends Project {
  roleUrls?: RoleUrl[]
}